src/state_machine/coordinator/fire.rs

```diff
@@ -638,28 +652,58 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
-                                let signer_public_key = &self.config.signer_public_keys[signer_id];
-                                let bad_signer_public_key =
-                                    &self.config.signer_public_keys[bad_signer_id];
+                                let Some(signer_key_ids) =
+                                    self.config.public_keys.signer_key_ids.get(signer_id)
+                                else {
+                                    warn!("No key IDs for signer_id {signer_id} DkgEnd");
+                                    continue;
+                                };
+                                let Some(signer_public_dsa_key) =
+                                    self.config.public_keys.signers.get(signer_id)
+                                else {
+                                    warn!("No public key for signer_id {signer_id} DkgEnd");
+                                    continue;
+                                };
+                                let signer_public_key = compute::point(signer_public_dsa_key)?;
+
+                                let Some(bad_signer_ecdsa_key) =
+                                    self.config.public_keys.signers.get(bad_signer_id)
+                                else {
+                                    warn!("Signer {signer_id} reported BadPrivateShare from invalid signer_id {bad_signer_id}, mark {signer_id} as malicious");
+                                    self.malicious_dkg_signer_ids.insert(*signer_id);
+                                    continue;
+                                };
+                                let bad_signer_public_key = compute::point(bad_signer_ecdsa_key)?;
                                 let mut is_bad = false;
 
                                 if bad_private_share.tuple_proof.verify(
-                                    signer_public_key,
-                                    bad_signer_public_key,
+                                    &signer_public_key,
+                                    &bad_signer_public_key,
                                     &bad_private_share.shared_key,
                                 ) {
                                     // verify at least one bad private share for one of signer_id's key_ids
                                     let shared_secret =
                                         make_shared_secret_from_key(&bad_private_share.shared_key);
 
-                                    let dkg_public_shares = &self.dkg_public_shares[bad_signer_id]
+                                    let Some(bad_dkg_public_shares) =
+                                        self.dkg_public_shares.get(bad_signer_id)
+                                    else {
+                                        warn!("Signer {signer_id} reported BadPrivateShare from signer {bad_signer_id} who didn't send public shares , mark {signer_id} as malicious");
```

Comment:
very nit (to be added in some other random PR), also below
```suggestion
                                        warn!("Signer {signer_id} reported BadPrivateShare from signer {bad_signer_id} who didn't send public shares, mark {signer_id} as malicious");
```